### ISSUE TITLE
[BUG]: incorrectly parsing string parameter containing regular expression pattern

### ISSUE THREAD

**Issue by luckeyca** [microsoft/azure-pipelines-tasks#20274]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

Bash@3

### Task version

3

### Issue Description

If a string parameter is regular expression pattern, even without single quote, bash task incorrectly parsed the string. Please see the technical details in repro steps. powershell task doesn't have this problem and shown below as comparison to bash task.

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Ubuntu 22.04, Oracle Linux 8.9

### Relevant log output

```shell
N/A. Please see the repro steps for all the technical details.
```

### Full task logs with system.debug enabled

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Repro steps

```yml
pipeline script:

trigger:
- main

pool:
  vmImage: ubuntu-latest

parameters:
- name: REGEX_BASE_IMAGE
  type: string
  displayName: Regular expression for OCI Platform Image
  default: '^Oracle-Linux-8\\.\\d{1,2}-2[0-9][0-9][0-9]\\.\\d+'
- name: REGEX_GOLDEN_IMAGE
  type: string
  displayName: Regular expression for Golden Image
  default: '^Oracle-Linux-8-Golden-Image-\\d{1,3}\\.\\d{1,3}.\\d{1,3}-*$'

steps:
- task: PowerShell@2
  inputs:
    targetType: inline
    script:
      Write-Host "${{ parameters.REGEX_BASE_IMAGE }}"
      Write-Host "${{ parameters.REGEX_GOLDEN_IMAGE }}"
  displayName: powershell regex string parameter
- task: Bash@3
  inputs:
    targetType: 'inline'
    script: |
      echo "${{ parameters.REGEX_BASE_IMAGE }}"
      echo "${{ parameters.REGEX_GOLDEN_IMAGE }}"
  displayName: bash regex string parameter
``` 

Expected Results:

Since the parameter strings are quoted with single quote, both powershell and bash task should output the strings as is without any parsing.

Actual Results: powershell task process the string correctly, but bash task removed the backslash. see details below
Output from powershell task with correct results.
```
^Oracle-Linux-8\\.\\d{1,2}-2[0-9][0-9][0-9]\\.\\d+ 
^Oracle-Linux-8-Golden-Image-\\d{1,3}\\.\\d{1,3}.\\d{1,3}-*$
```

Output from bash task with missing backslashes:
```
^Oracle-Linux-8\.\d{1,2}-2[0-9][0-9][0-9]\.\d+
^Oracle-Linux-8-Golden-Image-\d{1,3}\.\d{1,3}.\d{1,3}-*$
```
```

**Comment by ismayilov-ismayil** [comment]
Hi @luckeyca, thanks for reporting! We are working on more prioritized issues at the moment, but will get back to this one soon.

**Comment by ismayilov-ismayil** [comment]
Hi @luckeyca,
After investigation I found out that there is not bug regarding parsing. 
By default "Powershell" and "Bash" works a bit different. use this sample on Powerrshell and Bash terminal:
`Powershell:`
```
$TESTDATA="test\\data"; Write-Host $TESTDATA
```
output: `test\\data`
`Bash`
```
TESTDATA="test\\data"; echo $TESTDATA
```
output: `test\data`
as you can see Powershell doesn't use \ as escape symbol even it's double quoted.

When you set parameter in the yaml file, it's calculated before any task execution and replace that variable in place, instead of runtime. That means, when Bash or Powershell task starts the variable is already converted to string. 

This is content of bash script which will be executed:
![image](https://github.com/user-attachments/assets/d3ed13e5-0760-4072-8348-dd538fade901)

As shown in the screenshot regardless the quote around parameter we have double quote in the script, because we used double quote in the bash task content.

You can resolve this issue by replacing double quote in the script content like this:
 `echo  '${{ parameters.REGEX_BASE_IMAGE }}'`

**Comment by luckeyca** [comment]
Hi @ismayilov-ismayil you are right. I should use single quote across the board. thanks.
